plumgrid_playbooks/roles/plumgrid_director/tasks/main.yml

```diff
@@ -14,26 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Extract License from file
 - name: Extract License
   set_fact:
     pg_license: "{{ lookup('file', pg_license_path) }}"
 
-# Modify template files
-- name: Setup Keepalived Config
+- name: Setup Keepalived and Nginx config
   template:
-    src: keepalived.conf
-    dest: /var/lib/libvirt/filesystems/plumgrid-data/conf/etc/keepalived.conf
-
-- name: Setup nginx Config
-  template:
-    src: default.conf
-    dest: /var/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx.conf
-
-- name: Restart PLUMgrid service
-  service:
-    name: plumgrid
-    state: restarted
+    src: "{{ item.src }}"
+    dest: "{{ plumgrid_data_dir }}/{{ item.dest }}"
+  with_items:
```

Comment:
@arbabnazar in plumgrid_playbooks/roles/plumgrid_director/tasks/main.yml line:29 Service does not restart actually unless a separate task defined:
- name: Restart PLUMgrid
   service: name=plumgrid state=restarted
   notify: Restart PLUMgrid Service

The task logs ran with your commit are as follows:
TASK [plumgrid_director : Setup Keepalived and Nginx config] *******************
task path: /opt/plumgrid-ansible-standalone/plumgrid_playbooks/roles/plumgrid_director/tasks/main.yml:21
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/
tmp/ansible-tmp-1468853685.54-101520377369570 `" && echo ansible-tmp-1468853685.54-101520377369570="` echo $HOME/.ansible/tmp/ansible-tmp-1
468853685.54-101520377369570 `" ) && sleep 0'"'"''
<192.168.50.10> PUT /tmp/tmpOtGdhO TO /root/.ansible/tmp/ansible-tmp-1468853685.54-101520377369570/stat
<192.168.50.10> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthen
tication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o Conne
ctTimeout=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[192.168.50.10]'
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGE
S=en_US.UTF-8 /usr/bin/python && sleep 0'"'"''
<192.168.50.10> PUT /tmp/tmpEgeMsf TO /root/.ansible/tmp/ansible-tmp-1468853685.54-101520377369570/file
<192.168.50.10> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthen
tication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o Conne
ctTimeout=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[192.168.50.10]'
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGE
S=en_US.UTF-8 /usr/bin/python && sleep 0'"'"''
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-146885
3685.54-101520377369570/ > /dev/null 2>&1 && sleep 0'"'"''
ok: [director1] => (item={u'dest': u'conf/etc/keepalived.conf', u'src': u'keepalived.conf.j2'}) => {"changed": false, "diff": {"after": {"p
ath": "/var/lib/libvirt/filesystems/plumgrid-data/conf/etc/keepalived.conf"}, "before": {"path": "/var/lib/libvirt/filesystems/plumgrid-dat
a/conf/etc/keepalived.conf"}}, "gid": 0, "group": "root", "invocation": {"module_args": {"backup": null, "content": null, "delimiter": null
, "dest": "/var/lib/libvirt/filesystems/plumgrid-data/conf/etc/keepalived.conf", "diff_peek": null, "directory_mode": null, "follow": true,
 "force": false, "group": null, "mode": null, "original_basename": "keepalived.conf.j2", "owner": null, "path": "/var/lib/libvirt/filesyste
ms/plumgrid-data/conf/etc/keepalived.conf", "recurse": false, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype"
: null, "seuser": null, "src": null, "state": null, "validate": null}}, "item": {"dest": "conf/etc/keepalived.conf", "src": "keepalived.con
f.j2"}, "mode": "0644", "owner": "root", "path": "/var/lib/libvirt/filesystems/plumgrid-data/conf/etc/keepalived.conf", "size": 435, "state
": "file", "uid": 0}
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root                                                                          [191/9625]
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/
tmp/ansible-tmp-1468853685.84-224716263980770 `" && echo ansible-tmp-1468853685.84-224716263980770="` echo $HOME/.ansible/tmp/ansible-tmp-1
468853685.84-224716263980770 `" ) && sleep 0'"'"''
<192.168.50.10> PUT /tmp/tmpylPXtJ TO /root/.ansible/tmp/ansible-tmp-1468853685.84-224716263980770/stat
<192.168.50.10> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthen
tication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o Conne
ctTimeout=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[192.168.50.10]'
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGE
S=en_US.UTF-8 /usr/bin/python && sleep 0'"'"''
<192.168.50.10> PUT /tmp/tmpQs9KT0 TO /root/.ansible/tmp/ansible-tmp-1468853685.84-224716263980770/file
<192.168.50.10> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthen
tication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o Conne
ctTimeout=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[192.168.50.10]'
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGE
S=en_US.UTF-8 /usr/bin/python && sleep 0'"'"''
<192.168.50.10> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.50.10> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthenticati
on=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTime
out=120 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 192.168.50.10 '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-146885
3685.84-224716263980770/ > /dev/null 2>&1 && sleep 0'"'"''
ok: [director1] => (item={u'dest': u'conf/pg/nginx.conf', u'src': u'default.conf.j2'}) => {"changed": false, "diff": {"after": {"path": "/v
ar/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx.conf"}, "before": {"path": "/var/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx
.conf"}}, "gid": 0, "group": "root", "invocation": {"module_args": {"backup": null, "content": null, "delimiter": null, "dest": "/var/lib/l
ibvirt/filesystems/plumgrid-data/conf/pg/nginx.conf", "diff_peek": null, "directory_mode": null, "follow": true, "force": false, "group": n
ull, "mode": null, "original_basename": "default.conf.j2", "owner": null, "path": "/var/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx
.conf", "recurse": false, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": null,
 "state": null, "validate": null}}, "item": {"dest": "conf/pg/nginx.conf", "src": "default.conf.j2"}, "mode": "0644", "owner": "root", "pat
h": "/var/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx.conf", "size": 4370, "state": "file", "uid": 0}